lib/snapshot/BUILD.gn

```diff
@@ -248,7 +248,8 @@ bin_to_linkable("platform_strong_dill_linkable") {
 # Builds gen_snapshot using the host toolchain then copies the resulting binary
 # to `gen_snapshot_armv7` or `gen_snapshot_arm64` depending on the target
 # platform.
-if (host_os == "mac" && (target_cpu == "arm" || target_cpu == "arm64")) {
+if (host_os == "mac" && target_os != "mac" &&
+    (target_cpu == "arm" || target_cpu == "arm64")) {
```

Comment:
Note that this target:
* wasn't built for x64 macOS builds (correct)
* was built for arm64 macOS builds (incorrect)

The create_macos_gen_snapshots target:
* was built for both x64 and arm64 builds (correct)

In a following patch, I'll be merging these two targets into a single target.